searchcore/src/vespa/searchcore/proton/bucketdb/checksumaggregator.cpp

```diff
@@ -0,0 +1,16 @@
+#include "checksumaggregators.h"
+
+namespace proton::bucketdb {
+
+std::unique_ptr<ChecksumAggregator>
+ChecksumAggregator::create(ChecksumType type, BucketChecksum seed) {
+    switch (type) {
+    case ChecksumType::LEGACY:
+        return std::make_unique<LegacyChecksumAggregator>(seed);
+    case ChecksumType::XXHASH64:
+        return std::make_unique<XXHChecksumAggregator>(seed);
+    }
+    return std::unique_ptr<ChecksumAggregator>();
```

Comment:
`abort()` instead?

Reply:
I wondered why I needed it, as the compiler should see that all possible codepaths will return.
I thought that was one the advantages to using 'enum class'.

Reply:
abort it is...